src/base.jl

```diff
@@ -24,16 +37,21 @@ $(SIGNATURES)
 
 Return Hermitian conjugate \$\\langle val| = |val\\rangle^\\dagger\$ of the ket with the same label.
 """
-bra(val::Int, dim::Int) = bra(ComplexF64, val, dim)
+bra(val::Int, dim::Int) = bra(Vector{ComplexF64}, val, dim)
 
-function ketbra(::Type{T}, valk::Int, valb::Int, dim::Int) where T<:Number
+function ketbra(::Type{T}, valk::Int, valb::Int, dim::Int) where T<:AbstractMatrix{<:Number}
     dim > 0 ? () : throw(ArgumentError("Vector dimension has to be nonnegative"))
     1 <= valk <= dim && 1 <= valb <= dim ? () : throw(ArgumentError("Ket and bra labels have to be smaller than operator dimension"))
-    ρ = zeros(T, dim, dim)
-    ρ[valk,valb] = one(T)
+    ρ = zero(T(undef, dim, dim))
```

Comment:
We should support the case of rectangular matrices. Add idim (bra dim), odim (ket dim).

Reply:
Fixed